control-plane/src/modules/service-definitions.ts

```diff
@@ -1,7 +1,13 @@
 import { and, eq } from "drizzle-orm";
+import NodeCache from "node-cache";
+import { z } from "zod";
 import * as data from "./data";
 import { backgrounded } from "./util";
-import { z } from "zod";
+
+const cache = new NodeCache({
```

Comment:
I wonder if we could cache function results in `ts-core` also based on the cache key, so they don't even hit the control-plane?

Reply:
Yeah definitely. I think this is doable. But to do that we have to ferry the service definition to the client. Or, just encode the cache config in the client call.